tests/e2e/k8s-dra-gpu-driver_test.go

```diff
@@ -0,0 +1,177 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package e2e
+
+import (
+	"context"
+	"time"
+
+	helm "github.com/mittwald/go-helm-client"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	v1Core "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/NVIDIA/k8s-test-infra/pkg/diagnostics"
+)
+
+// Actual test suite
+var _ = Describe("K8S DRA GPU Driver", Ordered, func() {
+	AfterEach(func(ctx context.Context) {
+		// Run diagnostic collector if test failed
+		if CurrentSpecReport().Failed() {
+			var err error
+			diagnosticsCollector, err = diagnostics.New(
+				diagnostics.WithNamespace(testNamespace.Name),
+				diagnostics.WithArtifactDir(LogArtifactDir),
+				diagnostics.WithKubernetesClient(clientSet),
+				diagnostics.WithObjects(CollectLogsFrom...),
+			)
+			Expect(err).NotTo(HaveOccurred())
+
+			err = diagnosticsCollector.Collect(ctx)
+			Expect(err).NotTo(HaveOccurred())
+		}
+	})
+
+	When("deploying the K8S DRA Driver GPU via Helm", func() {
+		It("should be successful", func(ctx context.Context) {
```

Comment:
In the spirit of maximizing for conclusiveness... what would a corresponding `helm install` command line look like?

What is this helm chart really installed from? I saw an environment variable `HELM_CHART`. Is this meant to carry a path to a directory? Or to a tarball? If that is really the source of the Helm chart to be deployed maybe we can rename the variable to make that clearer, and also add a code comment here and there clarifying which Helm chart this really tests :).


I am not sure if I prefer a programmatic Helm client vs a CLI command executed.

Somewhere, we should test

Reply:
This could translate to aprox ~=

```bash
helm install nvidia-dra-driver-gpu $HELM_CHART \
    --version="$E2E_IMAGE_TAG" \
    --create-namespace \
    --namespace nvidia-dra-driver-gpu \
    --set nvidiaDriverRoot=/ \
    --set nvidiaCtkPath=/usr/bin/nvidia-ctk \
    --set gpuResourcesEnabledOverride=true
```

The idea is to deploy the helm chart from the `deployments/helm` folder during CI, but the HELM_CHART env var leaves the option during manual run to point to a diff location, that could be a tar.gz file.

100% agree with you, the idea of this initial E2E is to test https://github.com/NVIDIA/k8s-dra-driver-gpu/discussions/249 in a programmatic way